### ISSUE TITLE
AzureRm managed private endpoint prevents terraform plan after endpoint approval 

### ISSUE THREAD

**Issue by dps3ven** [hashicorp/terraform-provider-azurerm#21043]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.7

### AzureRM Provider Version

3.47.0

### Affected Resource(s)/Data Source(s)

azurerm_storage_account, azurerm_data_factory_managed_private_endpoint

### Terraform Configuration Files

```hcl
provider "azurerm" {
  features {}
}

data "azurerm_resource_group" "resource_group" {
  name = local.resource_group_name
}

resource "random_string" "random" {
  length  = 2
  special = false
  upper   = false
}


resource "azurerm_data_factory" "example" {
  name                            = format("example-adf-%s", random_string.random.result)
  location                        = data.azurerm_resource_group.resource_group.location
  resource_group_name             = data.azurerm_resource_group.resource_group.name
  managed_virtual_network_enabled = true
}

resource "azurerm_storage_account" "example" {
  name                     = format("example-sta-%s", random_string.random.result)
  resource_group_name      = data.azurerm_resource_group.resource_group.name
  location                 = data.azurerm_resource_group.resource_group.location
  account_tier             = "Standard"
  account_replication_type = "GRS"
}

resource "azurerm_data_factory_managed_private_endpoint" "endpoint" {
  name               = "test-endpoint"
  data_factory_id    = azurerm_data_factory.example.id
  target_resource_id = azurerm_storage_account.example.id
  subresource_name   = "blob"
}
```

### Debug Output/Panic Output

```shell
2023-03-20T10:57:29.807-0500 [ERROR] vertex "azurerm_storage_account.example" error: reading static website for AzureRM Storage Account "example-sta-iw": accounts.Client#GetServiceProperties: Failure sending request: StatusCode=0 -- Original Error: context deadline exceeded
```

### Expected Behaviour

No changes. Your infrastructure matches the configuration.

### Actual Behaviour

│ Error: reading static website for AzureRM Storage Account "example-sta-iw": accounts.Client#GetServiceProperties: Failure sending request: StatusCode=0 -- Original Error: context deadline exceeded

### Steps to Reproduce

2. terraform plan (success)
3. Use azure portal to approve endpoint connection
4. terraform plan (failure)

### Important Factoids

none

### References

https://github.com/hashicorp/terraform-provider-azurerm/issues/13889

The linked issue was closed without a clear path for resolution.

**Comment by tombuildsstuff** [comment]
hi @dps3ven 

Taking a look through here this issue is tracked in #2977 and is unfortunately blocked on the Storage Service Team making the Resource Manager API feature-compatible with the Data Plane API.

Unfortunately the Resource Manager API for Storage doesn't contain enough information to be able to populate all fields - as such we have to call the Data Plane API which is only available without a Private Endpoint configured/Public Access Disabled. [More information can be found in this issue](https://github.com/hashicorp/terraform-provider-azurerm/issues/13889#issuecomment-952693965) - however unfortunately this is a duplicate of #2977, which is blocked on the Storage Service Team making these API's compatible - and as such I'm going to close this issue in favour of #2977, would you mind subscribing to that issue for updates?

Thanks!
